### PR TITLE
Limits and boundaries page

### PR DESCRIPTION
**Pull request by Alzimax**
Added the limits and boundaries page with sliders that automatically use put requests.

### CODE REVIEW

File: libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  title: string;
+  marks: SliderMarks;
+  boundariesData: IBoundaries | undefined;
+  type: 'temperature' | 'humidity' | 'co2';
+  max: number;
+}) => {
+  const [dataTemp, setTempData] = useState<ITempData | null>();
+  const [dataHum, setHumData] = useState<IHumData | null>();
+  const [dataCO2, setCO2Data] = useState<ICO2Data | null>();
+  //const [data, setData] = useState<IBoundaries | null>();
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx:24]: Should be removed

File: libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  const { trigger } = usePutRequest<unknown, IBoundaries>({
+    url: '/Terrarium/boundaries',
+    data: data,
+  });
+  const [message, setMessage] = useState<INotificationState>();
+  const handlePut = async () => {
+    console.log('Trigger');
+    try {
+      const response = await trigger();
+      console.log(response?.data);
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx:36]: Should be removed

File: libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  //const [data, setData] = useState<IBoundaries | null>();
+  const data = boundariesData;
+  const [api, contextHolder] = notification.useNotification();
+  const { trigger } = usePutRequest<unknown, IBoundaries>({
+    url: '/Terrarium/boundaries',
+    data: data,
+  });
+  const [message, setMessage] = useState<INotificationState>();
+  const handlePut = async () => {
+    console.log('Trigger');
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx:33]: Should be removed

File: libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    try {
+      const response = await trigger();
+      console.log(response?.data);
+      return response;
+    } catch (error) {
+      console.error(error);
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx:39]: Could be removed

File: libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }, [dataCO2, dataHum, dataTemp]);
+
+  useEffect(() => {
+    if (message) openNotification('bottomRight', message?.type);
+  }, [message]);
+
+  let defaultValue: [number, number] = [0, 50];
+
+  if (type === 'temperature') {
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/BoundariesSlider.tsx:110]: Doesn't feel good if a code isn't in a hook or component, should be probably refactored

**Review by EnBi81 — APPROVED** [review]
Looks good, would love to see some more test cases! It remains for the next sprint

File: libs/limits-and-boundaries/src/lib/components/Interfaces.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  co2Min: number;
+  co2Max: number;
+}
+
+interface INotificationState {
+  message: string;
+  type: NotificationType;
+}
+
+export { IBoundaries, ILimits, ITempData, IHumData, ICO2Data, INotificationState, NotificationType };
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/Interfaces.tsx:39]: I think you can also export them separately, which saves you one line of code :D but I think it's more like a personal choice

File: libs/limits-and-boundaries/src/lib/components/Limits.tsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
+import { notification, Skeleton, Slider } from 'antd';
+import React, { useEffect, useState } from 'react';
+import { style, marksTemp } from './MarksAndStyle';
+import { ILimits } from './Interfaces';
+import { ExtractValues } from './ExtractValues';
+import { NotificationPlacement } from 'antd/es/notification/interface';
+import { usePutRequest } from '@lizards-inc-fe/fetcher';
+
+type NotificationType = 'success' | 'info' | 'warning' | 'error';
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/Limits.tsx:9]: Already defined in interfaces, no need to for creating that again

File: libs/limits-and-boundaries/src/lib/components/Limits.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  const [limits, setLimits] = useState<ILimits | undefined>();
+
+  const handlePut = async () => {
+    console.log('TriggerLimits');
+    try {
+      const response = await trigger();
+      console.log(response?.data);
+      return response;
+    } catch (error) {
+      console.error(error);
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/Limits.tsx:32]: Could be removed

File: libs/limits-and-boundaries/src/lib/components/Limits.tsx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    data: limitsData,
+  });
+  const [message, setMessage] = useState<INotificationState>();
+  const [limits, setLimits] = useState<ILimits | undefined>();
+
+  const handlePut = async () => {
+    console.log('TriggerLimits');
+    try {
+      const response = await trigger();
+      console.log(response?.data);
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/Limits.tsx:29]: Should be removed

File: libs/limits-and-boundaries/src/lib/components/Limits.tsx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+  const [api, contextHolder] = notification.useNotification();
+  const { trigger } = usePutRequest<unknown, ILimits>({
+    url: '/Terrarium/limits',
+    data: limitsData,
+  });
+  const [message, setMessage] = useState<INotificationState>();
+  const [limits, setLimits] = useState<ILimits | undefined>();
+
+  const handlePut = async () => {
+    console.log('TriggerLimits');
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/Limits.tsx:26]: Should be removed

File: libs/limits-and-boundaries/src/lib/components/Limits.tsx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+      handlePut()
+        .then(response => {
+          if (response) {
+            setMessage({ message: 'Changes successfully saved!', type: 'success' });
+          } else {
+            setMessage({ message: 'Something went wrong!', type: 'error' });
+          }
+        })
+        .catch(error => {
+          console.error(error);
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/Limits.tsx:68]: Could be removed

File: libs/limits-and-boundaries/src/lib/components/MarksAndStyle.tsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+const marksCO2: SliderMarks = {
+  0: {
+    label: '0%',
+  },
+  100: {
+    label: '100%',
+  },
+};
+
+export { marksTemp };
```
Comment (apeTrackunit) [inline libs/limits-and-boundaries/src/lib/components/MarksAndStyle.tsx:43]: You can export the constants directly

`export const marksCO2: SliderMarks = {`

File: src/app/app.tsx [hunk 13]
```diff
@@ -6,9 +6,12 @@ import { HomeOutlined, TwitterOutlined } from '@ant-design/icons';
 import { Outlet, useNavigate } from 'react-router-dom';
 import { useEffect, useState } from 'react';
 import { MenuFoldOutlined, MenuUnfoldOutlined } from '@ant-design/icons';
+import {RoutingTable} from "@lizards-inc-fe/shared-components";
+import { SlidersOutlined } from '@ant-design/icons';
 import { NotificationCenter } from '@lizards-inc-fe/notifcation-center';
 
-export const App = () => {
+
+const App = () => {
```
Comment (apeTrackunit) [inline src/app/app.tsx:14]: What's the reason for removing the `export`?

**Pull request merged**